bldsys/cmake/template/gradle/local.properties.in

```diff
@@ -0,0 +1 @@
+sdk.dir=@ANDROID_SDK@
```

Comment:
Isn't it bad practice to include local.properties in source repo?  I thought this file is auto generated by AS.

Reply:
TBH, not sure. Rememeber our instructions on for non-AS based builds. i.e. straight from command line, in which case this is not generated.

Reply:
Right, I forgot about the instructions.  I use AS for any android building and CLion rest of the time; our instructions are less efficient than using IDEs.

Reply:
Given the only thing it's being used for is the location of the SDK and duplication from the ENV variable, why not just use the ENV variable directly in CMake and call it a day?  My concern is that If someone has a custom local.properties this might interfere with it?  (I honestly don't know if it would or not, locally we ban letting local.properties into source control).

Reply:
@gpx1000 But I don't need the sdk path in cmake. I need this in android scripts (gradle). `local.properties` would be bad if submitted in source control with hard-coded paths. In this case its only locally created to make android build happy.

Reply:
Technically, we could modify it to have cmake call gradle and cmake can call build directly, but that's a bit much extra.  The configure_file option will do what you're looking for, or simply adding it to the environment variables will allow gradle to not require the local.properties I believe?

Reply:
I'd be interested in what that environment vars deprecated message came from as I haven't seen it yet.

You definitely don't need both ANDROID_NDK_ROOT and ANDROID_NDK_HOME.  I see you've been developing Android a while as you have all of the old ways ;-).  So the bare minimum of what you need is ANDROID_HOME.  CCACHE can be found in other ways (i.e. if you're using GNU make then ANDROID_MAKE_CCACHE is more useful) however, in general I get CCACHE via the cmake build system identifying it and using it.

https://developer.android.com/studio/command-line/variables

I personally usually just set the ndkpath explicitly via android.ndkPath.  So on a CI environment, I'll grab the androidManager then use android.ndkPath to point at the NDK I might be using.  Which might or might not be the one installed via Android SDK Manager. (there's cases where I've had to have multiple versions installed).
https://developer.android.com/studio/projects/configure-agp-ndk

Reply:
ok I can verify that just leaving `ANDROID_HOME` and removing everything else including the `local.properties` still works. So I have got rid of those in the last commit. This simplifies things a bit.

Reply:
> I'd be interested in what that environment vars deprecated message came from as I haven't seen it yet.

I was thinking of this. I have seen these messages myself. https://github.com/KhronosGroup/Vulkan-Samples/pull/431

Reply:
I think those messages are saying correctly that ANDROID_NDK_ROOT is deprecated in favor of ANDROID_NDK_HOME.  ANDROID_HOME is enough if the ndk is found inside the SDK.  Which is by far the most useful and correct place for it to be.